flood_adapt/object_model/interface/measures.py

```diff
@@ -138,23 +157,88 @@ class GreenInfrastructureModel(HazardMeasureModel):
     """BaseModel describing the expected variables and data types of the "green infrastructure" hazard measure"""
 
     volume: UnitfulVolume = UnitfulVolume(value=0.0, units=UnitTypesVolume.m3)
-    height: UnitfulLength = UnitfulLength(value=0.0, units=UnitTypesLength.meters)
+    height: Optional[UnitfulLength] = None
     aggregation_area_type: Optional[str] = None
     aggregation_area_name: Optional[str] = None
-    percent_area: float = 100
+    percent_area: Optional[float] = None
+
+    @validator("type", always=True)
+    def validate_type(cls, type: HazardType, values: Any) -> HazardType:
+        if type not in [
+            HazardType.water_square,
+            HazardType.greening,
+            HazardType.total_storage,
+        ]:
+            raise ValueError(
+                "Type must be one of 'water_square', 'greening', or 'total_storage'"
+            )
+        return type
 
     @validator("volume")
     def validate_volume(cls, volume: UnitfulVolume, values: Any) -> UnitfulVolume:
         if volume.value <= 0:
             raise ValueError("Volume cannot be zero or negative")
         return volume
 
-    @validator("height")
-    def validate_height(cls, height: UnitfulLength, values: Any) -> UnitfulLength:
-        if height.value <= 0:
+    @validator("height", always=True)
+    def validate_height(
+        cls, height: Optional[UnitfulLength], values: Any
+    ) -> Optional[UnitfulLength]:
+        if values.get("type", "") == HazardType.total_storage:
+            if height is not None:
+                raise ValueError("Height cannot be set for total storage type measures")
+            return None  # Height is not required for total storage type measures
+        elif not isinstance(height, UnitfulLength):
+            raise ValueError("Height must be a UnitfulLength")
+        elif height.value <= 0:
             raise ValueError("Height cannot be zero or negative")
         return height
```

Comment:
We should use @model_validators for this!

Also some nitpicking, if height is not None -> if height: